tests/lean/importing/test_importing.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+if [ $# -ne 1 ]; then
+    echo "Usage: test_importing.sh [lean-executable-path]"
+    exit 1
+fi
+ulimit -s 8192
+LEAN=$1
+export LEAN_PATH=../../../library:.
+
+cat <<EOF > foo.lean
+run_cmd tactic.trace "foo"
+EOF
+rm -f foo.olean
+
+out=$($LEAN --make ./foo.lean 2>&1)
+if [[ $out != "foo" ]]; then
+    echo "-- Lean isn't building foo: got $out , expected foo"
+    exit 1
+fi
+
+touch foo.lean
+out=$($LEAN --make ./foo.lean 2>&1)
+if [[ $out != "" ]]; then
+    echo "-- Lean rebuilds foo on timestamp change: got $out , expected no output"
+    exit 1
+fi
+
+cat <<EOF > x.lean
+def x := 0
+run_cmd tactic.trace "x"
+EOF
+cat <<EOF > y.lean
+import x
+def y := x
+run_cmd tactic.trace "y"
+EOF
+cat <<EOF > z.lean
+import y
+def z := y
+run_cmd tactic.trace "z"
+EOF
+rm -f x.olean y.olean z.olean
+
+out=$($LEAN --make z.lean 2>&1 | tr -d '\n')
```

Comment:
```suggestion
out=$($LEAN -j0 --make z.lean 2>&1 | tr -d '\n')
```
The `-j0` option enables the deterministic single-threaded task manager.  Although you've already written the test files so that they can only be executed in one order.